OpenSim/Actuators/Millard2012EquilibriumMuscle.cpp

```diff
@@ -556,43 +556,38 @@ computeFiberEquilibrium(SimTK::State& s, bool solveForVelocity) const
 
-    try {
-        std::pair<StatusFromEstimateMuscleFiberState,
-                  ValuesFromEstimateMuscleFiberState> result =
-            estimateMuscleFiberState(activation, pathLength, pathSpeed,
-                tol, maxIter, solveForVelocity);
-
-        switch(result.first) {
-
-        case StatusFromEstimateMuscleFiberState::Success_Converged:
-            setActuation(s, result.second["tendon_force"]);
-            setFiberLength(s, result.second["fiber_length"]);
-            break;
-
-        case StatusFromEstimateMuscleFiberState::Warning_FiberAtLowerBound:
-            log_warn("Millard2012EquilibriumMuscle static solution: '{}' is "
-                   "at its minimum fiber length of {}.",
-                   getName(), result.second["fiber_length"]);
-            setActuation(s, result.second["tendon_force"]);
-            setFiberLength(s, result.second["fiber_length"]);
-            break;
-
-        case StatusFromEstimateMuscleFiberState::Failure_MaxIterationsReached:
-            // Report internal variables and throw exception.
-            std::ostringstream ss;
-            ss << "\n  Solution error " << abs(result.second["solution_error"])
-               << " exceeds tolerance of " << tol << "\n"
-               << "  Newton iterations reached limit of " << maxIter << "\n"
-               << "  Activation is " << activation << "\n"
-               << "  Fiber length is " << result.second["fiber_length"] << "\n";
-            OPENSIM_THROW_FRMOBJ(MuscleCannotEquilibrate, ss.str());
-            break;
-        }
+    const MuscleStateEstimate result =
+        estimateMuscleFiberState(
+            activation,
+            pathLength,
+            pathSpeed,
+            tol,
+            maxIter,
+            solveForVelocity);
+
+    if (result.status ==
+        MuscleStateEstimate::Status::Failure_MaxIterationsReached) {
+            std::ostringstream oss;
+            oss << "Failed to compute muscle equilibrium state:\n"
+                << "    Solution error " << std::abs(result.solutionError)
+                << " exceeds tolerance of " << tol << "\n"
+                << "    Newton iterations reached limit of " << maxIter << "\n"
+                << "    Activation is " << activation << "\n"
+                << "    Fiber length is " << result.fiberLength << "\n";
+            OPENSIM_THROW_FRMOBJ(MuscleCannotEquilibrate, oss.str());
+    }
 
-    } catch (const std::exception& x) {
-        OPENSIM_THROW_FRMOBJ(MuscleCannotEquilibrate,
-            "Internal exception encountered.\n" + std::string{x.what()});
+    if (result.status ==
+        MuscleStateEstimate::Status::Warning_FiberAtLowerBound) {
+            log_warn(
+                "Millard2012EquilibriumMuscle static solution: '{}' is at its"
+                    "minimum fiber length of {}.",
+                getName(),
+                result.fiberLength);
     }
+
+    setActuation(s, result.tendonForce);
```

Comment:
Previously, the `setActuation()` and `setFiberLength()` lines were not executed if `Failure_MaxIterationsReached`, so this PR changes the behavior of the method. Without a deeper investigation, I don't know whether there are consequences to changing this behavior.

Reply:
The `OPENSIM_THROW_FRMOBJ` throws an exception, which prevents anything thereafter being reached. This should still prevent `setActuation()` and `setFiberLength()` from being reached in that case.

Reply:
I thought this exception might have been caught and handled using another strategy but this doesn't seem to be the case.

Reply:
I'm not sure I follow.
Previously, if max iterations was reached and there was no convergence, we threw an exception:
```
       case StatusFromEstimateMuscleFiberState::Failure_MaxIterationsReached:
            // Report internal variables and throw exception.
            std::ostringstream ss;
            ...
            OPENSIM_THROW_FRMOBJ(MuscleCannotEquilibrate, ss.str());
            break;
```
this exception was caught, but in the `catch` block it threw another exception:
```
    } catch (const std::exception& x) {
        OPENSIM_THROW_FRMOBJ(MuscleCannotEquilibrate,
            "Internal exception encountered.\n" + std::string{x.what()});
```

I thought this would simplify to throwing an exception in case of no-convergence.
From your comment, I'm not sure if you agree or not :)

Reply:
Looks good… IIRC at some point there were exceptions in this class that were thrown and then caught internally to proceed with some default behavior. Not the case here 😌